realm/realm-library/src/androidTest/java/io/realm/RealmAsyncQueryTests.java

```diff
@@ -1864,153 +1087,11 @@ public void doInBackground(Realm realm) {
         });
     }
 
-    // Make sure we don't get the run into the IllegalStateException
-    // (Caller thread behind the worker thread)
-    // Scenario:
-    // - Caller thread is in version 1, start an asyncFindFirst
-    // - Another thread advance the Realm, now the latest version = 2
-    // - The worker thread should query against version 1 not version 2
-    // otherwise the caller thread wouldn't be able to import the result
-    // - The notification mechanism will guarantee that the REALM_CHANGE triggered by
-    // the background thread, will update the caller thread (advancing it to version 2)
-    @Test
-    @RunTestInLooperThread
-    public void testFindFirstUsesCallerThreadVersion() throws Throwable {
-        final CountDownLatch signalClosedRealm = new CountDownLatch(1);
-
-        populateTestRealm(looperThread.realm, 10);
-        Realm.asyncTaskExecutor.pause();
-
-        final AllTypes firstAsync = looperThread.realm.where(AllTypes.class).findFirstAsync();
-        looperThread.keepStrongReference.add(firstAsync);
-        firstAsync.addChangeListener(new RealmChangeListener<AllTypes>() {
-            @Override
-            public void onChange(AllTypes object) {
-                assertNotNull(firstAsync);
-                assertEquals("test data 0", firstAsync.getColumnString());
-                looperThread.testComplete(signalClosedRealm);
-            }
-        });
-
-        // advance the background Realm
-        new Thread() {
-            @Override
-            public void run() {
-                Realm bgRealm = Realm.getInstance(looperThread.realmConfiguration);
-                // Advancing the Realm without generating notifications
-                bgRealm.sharedRealm.beginTransaction();
-                bgRealm.sharedRealm.commitTransaction();
-                Realm.asyncTaskExecutor.resume();
-                bgRealm.close();
-                signalClosedRealm.countDown();
-            }
-        }.start();
-    }
-
-    // Test case for https://github.com/realm/realm-java/issues/2417
-    // Ensure that a UnreachableVersion exception during handover doesn't crash the app or cause a segfault.
-    @Test
-    @UiThreadTest
-    public void badVersion_findAll() throws NoSuchFieldException, IllegalAccessException {
```

Comment:
The original case is for test `load()` if async query cannot handover the results. Without calling `load()`, the test case will be meaningless. But current `load()` is only setting a flag and will always return true. Those test cases are not needed.